cw/capture.py

```diff
@@ -1127,118 +1128,261 @@ def capture_otbn_vertical(ot, ktp, fw_bin, pll_frequency, capture_cfg, device_cf
     # Initialize some curve-dependent parameters.
     if capture_cfg["curve"] == 'p256':
         curve_order_n = 0xffffffff00000000ffffffffffffffffbce6faada7179e84f3b9cac2fc632551
-        # key_bytes = 256 // 8
+        key_bytes = 256 // 8
         seed_bytes = 320 // 8
+        modinv_share_bytes = 320 // 8
+        modinv_mask_bytes = 128 // 8
     else:
         # TODO: add support for P384
-        raise NotImplementedError(f'Curve {capture_cfg["curve"]} is not supported')
-
-    # Check the lengths in the key/plaintext generator. In this case, "key"
-    # means seed and "plaintext" means mask.
-    if ktp.keyLen() != seed_bytes:
-        raise ValueError(f'Unexpected seed length: {ktp.keyLen()}.\n'
-                         f'Hint: set key len={seed_bytes} in the configuration file.')
-    if ktp.textLen() != seed_bytes:
-        raise ValueError(f'Unexpected mask length: {ktp.textLen()}.\n'
-                         f'Hint: set plaintext len={seed_bytes} in the configuration file.')
-
-    # Seed the RNG and generate a random fixed seed for all traces of the
-    # keygen operation.
-    seed_fixed = ktp.next_key()
-    print(f'fixed seed = {seed_fixed.hex()}')
-    if len(seed_fixed) != seed_bytes:
-        raise ValueError(f'Fixed seed length is {len(seed_fixed)}, expected {seed_bytes}')
-
-    # Expected key is `seed mod n`, where n is the order of the curve and
-    # `seed` is interpreted as little-endian.
-    expected_fixed_key = int.from_bytes(seed_fixed, byteorder='little') % curve_order_n
+        raise NotImplementedError(
+            f'Curve {capture_cfg["curve"]} is not supported')
 
     # register ctrl-c handler to not lose already recorded traces if measurement is aborted
     signal.signal(signal.SIGINT, partial(abort_handler, project))
 
-    sample_fixed = 1
-    # Loop to collect each power trace
-    for _ in tqdm(range(capture_cfg["num_traces"]), desc='Capturing',
-                  ncols=80):
+    if capture_cfg["app"] == 'keygen':
```

Comment:
Thanks @abdullahvarici for pointing this out. This is also something I thought about while writing the code. And yes, there definitely will be merge conflicts with Vladimir's code. But if I modify my PR, and split up the code, merging will get even more complicated. So maybe we should somehow try to solve the merge conflicts between Vladimir and me, and deal with code restructuring afterwards.
Anyway, we should discuss this in the SCA WG meeting, because initially we decided to include everything into one function.

Reply:
Just a short follow-up comment to have this documented here as well.
In the SCA WG meeting we decided to deal with the merge conflicts, and then afterwards discuss on how to structure the code.